src/test/ui/consts/const-eval/double_check2.rs

```diff
@@ -1,7 +1,9 @@
 // check-pass
 
 // This test exhibits undefined behavior, but it is impossible to prevent generally during
-// const eval, even if possible to prevent in the cases here. The reason it's impossible in general
+// const eval, even if it could be prevented in the cases here if we added expensive and
+// complex checks in rustc.
+// The reason it's impossible in general
```

Comment:
I still don't agree that this is impossible in general.

Reply:
right. This reminds me, we should probably replace the `const_eval_raw` query in miri to run full miri during const eval so we catch UB during const eval.

Reply:
> we should probably replace the const_eval_raw query in miri to run full miri during const eval so we catch UB during const eval.

Hm, I am not sure we want that TBH. It would seem rather surprising to me if CTFE behaved differently in Miri than normally. Const-UB is somewhat different from runtime-UB, and Miri does detect runtime-UB caused by consts having wrong results due to const-UB.